### PR TITLE
Removing ts-ignores

### PR DESCRIPTION
**Pull request by krokhmaliuk-virtru**

### CODE REVIEW

File: lib/facade/FileClient.ts [hunk 1]
```diff
@@ -36,9 +36,7 @@ export class FileClient {
     source: ReadableStream | Buffer | string | ArrayBuffer,
```
Comment (ivanovSPvirtru) [inline lib/facade/FileClient.ts:36]: We still should probably create NodeStream interface and add here at least for readability (and add comment why we not using node stream as type

File: lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts [hunk 2]
```diff
@@ -28,7 +28,31 @@ LWExNjctZWY5OGQyY2NmNjdhIn2avSz7nTV08u+z0lNoOax2ZSWlNtycmvQLS4zHNJn/2i8E1p+KGUx+
...
+  symmetricPayloadConfig: {
+    hasSignature: boolean;
+    signatureCurveName: number;
+    symmetricCipher: number;
+  };
+  policy: {
+    type: PolicyTypeEnum;
+    content: string[];
+    binding: string[];
+  };
```
Comment (ivanovSPvirtru) [inline lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts:51]: Its in mutiple places. Make sence to make one Policy type

File: lib/facade/FileClient.ts [hunk 3]
```diff
@@ -36,9 +36,7 @@ export class FileClient {
...
     params: EncryptParamsBuilder | DecryptParamsBuilder
   ) {
-    // @ts-ignore
-    if (source.pipe !== undefined || source instanceof ReadableStream) {
```
Comment (obidenko) [inline lib/facade/FileClient.ts:40]: source?.pipe   ?

File: lib/src/__fixtures__/nanotdf-spec-embedded-example.ts [hunk 4]
```diff
@@ -28,7 +28,31 @@ GtnSAeLvGsARKzdyTF9PpArXAc9AbC7VnbVyA7OMpPF0YFQDQHcAwtHIyzVZNSzarQdcChszFMezBVRf
 3egvvoM+IgjYoK1C
 `;
 
-export const header = {
+export interface Header {
```
Comment (obidenko) [inline lib/src/__fixtures__/nanotdf-spec-embedded-example.ts:31]: We can reuse this interface

File: lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts [hunk 5]
```diff
@@ -28,7 +28,31 @@ LWExNjctZWY5OGQyY2NmNjdhIn2avSz7nTV08u+z0lNoOax2ZSWlNtycmvQLS4zHNJn/2i8E1p+KGUx+
...
+  kas: {
+    protocol: number;
+    length: number;
+    body: string;
+  };
+  eccBindingMode: {
+    useECDSABinding: boolean;
+    ephemeralCurveName: number;
+  };
+  symmetricPayloadConfig: {
```
Comment (obidenko) [inline lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts:42]: we could make it as interface also and reuse

File: lib/tdf3/src/utils/aws-lib-storage/chunker.ts [hunk 6]
```diff
@@ -7,9 +7,7 @@ import { BodyDataTypes } from './types';
 export const getChunk = (data: BodyDataTypes, partSize: number) => {
   if (data instanceof Buffer) {
     return getChunkBuffer(data, partSize);
-    // @ts-ignore
-  } else if (data.pipe) {
-    // @ts-ignore
+  } else if (Object.prototype.hasOwnProperty.call(data, 'pipe')) {
```
Comment (obidenko) [inline lib/tdf3/src/utils/aws-lib-storage/chunker.ts:10]: data?.pipe as option

File: lib/tdf3/src/utils/aws-lib-storage/chunks/getDataReadable.ts [hunk 7]
```diff
@@ -1,5 +1,4 @@
-// @ts-ignore
-export async function* getDataReadable(data): AsyncGenerator<Buffer> {
+export async function* getDataReadable(data: any): AsyncGenerator<Buffer> {
```
Comment (obidenko) [inline lib/tdf3/src/utils/aws-lib-storage/chunks/getDataReadable.ts:1]: Can we get rid of "any" ?

File: lib/tdf3/src/utils/index.ts [hunk 8]
```diff
@@ -8,8 +8,7 @@ export function inBrowser(): boolean {
   return typeof window !== 'undefined';
 }
 
-// @ts-ignore
-export async function streamToBuffer(stream) {
+export async function streamToBuffer(stream: any) {
```
Comment (obidenko) [inline lib/tdf3/src/utils/index.ts:11]: any - could we have some type here?

File: lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts (hunk 2 above)
Comment (krokhmaliuk-virtru) [reply lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts:51]: Added new common types for reusability

File: lib/facade/FileClient.ts (hunk 1 above)
Comment (krokhmaliuk-virtru) [reply lib/facade/FileClient.ts:36]: I've added new "InputSource" type for this. 
I suppose it's neither Stream actually, looks more like input data source
Let me know if i'm wrong

File: lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts (hunk 2 above)
Comment (krokhmaliuk-virtru) [reply lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts:51]: Added common types as well

File: lib/facade/FileClient.ts (hunk 3 above)
Comment (krokhmaliuk-virtru) [reply lib/facade/FileClient.ts:40]: Actual problem is that TS doesn't understand what type is it.  So with optional operator also will get error warn

File: lib/src/__fixtures__/nanotdf-spec-embedded-example.ts (hunk 4 above)
Comment (krokhmaliuk-virtru) [reply lib/src/__fixtures__/nanotdf-spec-embedded-example.ts:31]: Added file with common types

File: lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts (hunk 5 above)
Comment (krokhmaliuk-virtru) [reply lib/src/__fixtures__/nanotdf-spec-plain-embedded-example.ts:42]: same as above

File: lib/tdf3/src/utils/aws-lib-storage/chunker.ts (hunk 6 above)
Comment (krokhmaliuk-virtru) [reply lib/tdf3/src/utils/aws-lib-storage/chunker.ts:10]: > Actual problem is that TS doesn't understand what type is it. So with optional operator also will get error warn

x2

File: lib/tdf3/src/utils/aws-lib-storage/chunks/getDataReadable.ts (hunk 7 above)
Comment (krokhmaliuk-virtru) [reply lib/tdf3/src/utils/aws-lib-storage/chunks/getDataReadable.ts:1]: will be cover in next tickets

File: lib/src/types/index.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  };
+  symmetricPayloadConfig: {
+    hasSignature: boolean;
+    signatureCurveName: number;
+    symmetricCipher: number;
+  };
+  ephemeralPublicKey: string[];
+};
+
+type Policy = {
```
Comment (ivanovSPvirtru) [inline lib/src/types/index.ts:24]: image
Already have this type. Check other types for duplecations, theres a lot in models defined already

File: lib/tdf3/src/ciphers/aes-gcm-cipher.ts [hunk 10]
```diff
@@ -7,16 +7,15 @@ import type { CryptoService, DecryptResult, EncryptResult } from '../crypto/decl
...
+function processGcmPayload(buffer: Buffer) {
   // Read the 12 byte IV from the beginning of the stream
   const payloadIv = Binary.fromBuffer(buffer.slice(0, 12));
 
   // Slice the final 16 bytes of the buffer for the authentication tag
   const payloadAuthTag = Binary.fromBuffer(buffer.slice(-16));
 
   return {
-    payload: Binary.fromBuffer(buffer.slice(12, -16)),
+    payload: Binary.fromBuffer(buffer.slice(12, -16)) as unknown as Buffer,
```
Comment (ivanovSPvirtru) [inline lib/tdf3/src/ciphers/aes-gcm-cipher.ts:18]: Lets walk trough this. It looks like Binary type.

File: cli/src/cli.ts [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import yargs from 'yargs';
 import { readFile, stat, writeFile } from 'fs/promises';
 import { hideBin } from 'yargs/helpers';
-// @ts-ignore
+//@ts-ignore
```
Comment (obidenko) [inline cli/src/cli.ts:4]: can we get rid of it?

**Review by ivanovSPvirtru — APPROVED** [review]

**Review by ivanovSPvirtru — DISMISSED** [review]

**Pull request closed without merging**